docs/proposals/nginx-extensions.md

```diff
@@ -0,0 +1,964 @@
+- [`otel_trace_context`](https://nginx.org/en/docs/ngx_otel_module.html#otel_trace_context)
+- [`otel_span_name`](https://nginx.org/en/docs/ngx_otel_module.html#otel_span_name)
+- [`otel_span_attr`](https://nginx.org/en/docs/ngx_otel_module.html#otel_span_attr)
+
+#### Future Work
+
+Add support for:
+
+- OTEL logging
+- Metrics
+
+#### Alternatives
+
+- Combine with OTEL settings in Gateway Settings for one OTEL Policy: Rather than splitting tracing across two Policies, we could create a single tracing Policy. The issue with this approach is that some tracing settings -- such as exporter endpoint -- should be restricted to Cluster Operators, while settings like attributes should be available to Application Developers. If we combine these settings, RBAC will not be sufficient to restrict access across the settings. We will have to disallow certain fields based on the resource the Policy is attached to. This is a bad user experience.
+- Inherited Policy: An Inherited Policy would be useful if there is a use case for the Cluster Operator enforcing or defaulting the OTEL tracing settings included in this policy.
+
+### Proxy Settings
+
+_Extension type:_ Inherited Policy
+
+_Resource type:_ CRD
+
+_Role(s):_ Cluster Operator, Application Developer
+
+_Extension point:_ Gateway, HTTPRoute, HTTPRoute Rule
+
+_NGINX context(s):_ http, server, location
+
+Features:
+
+- Proxy buffering
+- Connection timeouts
+- Next upstream retries
+
+NGINX directives:
+
+- [`proxy_connect_timeout`](https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_connect_timeout)
+- [`proxy_read_timeout`](https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_read_timeout)
+- [`proxy_send_timeout`](https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_send_timeout)
+- [`proxy_next_upstream`](https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_next_upstream)
+- [`proxy_next_upstream_timeout`](https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_next_upstream_timeout)
+- [`proxy_next_upstream_retries`](https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_next_upstream_tries)
+- [`proxy_buffering`](https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_buffering)
+
+#### Future Work
+
+- Add other `proxy_*` directives
+
+#### Alternatives
+
+- Direct Policy: If there's no strong use case for the Cluster Operator setting sane defaults for these settings, then we can use a Direct Policy. The Direct Policy could attach to an HTTPRoute or HTTPRoute Rule, and the NGINX contexts would be server and location.
+
+### Circuit Breaker/ Backup service
+
+_Extension type:_ Direct Policy/Filter
+
+_Resource type:_ CRD
+
+_Role(s):_ Application Developer
+
+_Extension point:_ Backend/HTTPRoute Rule
+
+_NGINX context(s):_ upstream
+
+Features:
+
+- Backup service/circuit breaker
+
+NGINX upstream server parameters:
+
+- [`backup`](https://nginx.org/en/docs/http/ngx_http_upstream_module.html#server)
```

Comment:
I think it's better to use the proposed above passive and active health checks for this, so that we won't have to deal with potentially conflicting settings

Reply:
Can you give me an example? I don't understand the suggestion.

Reply:
The thing that determines if primary backends fails is a combination of active and passive health checks. My advise here is to not bring fail_timeut and max_fails (which are passive health checks) into CB configuration, but rather use the proposed active and passive health checks configuration from the top of the document.